src/runtime/HalideBuffer.h

```diff
@@ -2408,7 +2408,7 @@ class Buffer {
              typename = typename std::enable_if<(d >= 0)>::type>
     HALIDE_ALWAYS_INLINE static void for_each_element_array_helper(int, const for_each_element_task_dim *t, Fn &&f, int *pos) {
         for (pos[d] = t[d].min; pos[d] <= t[d].max; pos[d]++) {
-            for_each_element_array_helper<d - 1>(0, t, std::forward<Fn>(f), pos);
```

Comment:
What was the issue here?

Reply:
On the second iteration, we try to reuse `f` -- that's a use-after-move.